opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -3692,49 +3694,55 @@ namespace Opm
     MultisegmentWell<TypeTag>::
     pressureDropSpiralICD(const int seg) const
     {
-        // TODO: We have to consider the upwinding here
         const SICD& sicd = *segmentSet()[seg].spiralICD();
 
-        const std::vector<EvalWell>& phase_fractions = segment_phase_fractions_[seg];
-        const std::vector<EvalWell>& phase_viscosities = segment_phase_viscosities_[seg];
+        const int seg_upwind = upwinding_segments_[seg];
+        // WARNING
+        // We disregard the derivatives from the upwind fractions and viscosities to make sure derivatives
+        // wrt. to different segments dont get mixed.
```

Comment:
Could it be an idea to only discard derivatives if the upwind segment is not the current one?

Reply:
Then we lose some derivatives. Is that what you are thinking about?

Because the upwinding segment is also in the system.

Reply:
I agree we should go into that direction, but think that should be done in another PR.

Reply:
Ok with me.